File: backend/src/main/java/com/andrew/hdss/dtos/BatchResponse.java
package com.andrew.hdss.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BatchResponse<T> {
    private Integer size;
    private Integer page;
    private Long totalElements;
    private Integer totalPages;
    private List<T> data;
}

File: android/app/src/main/java/com/andrew/hdss/dtos/BatchResponse.kt
package com.andrew.hdss.dtos

import kotlinx.serialization.Serializable

@Serializable
data class BatchResponse<T>(
val size: Int? = null,
val page: Int? = null,
val totalElements: Long? = null,
val totalPages: Int? = null,
val data: MutableList<T> = mutableListOf()
)
